.github/workflows/deploy-ghpages-latest-stable.yml

```diff
@@ -0,0 +1,151 @@
+name: Deploy Sphinx documentation
+
+on:
+
+  workflow_call:
+    inputs:
+        python-version:
+          required: true
+          type: string
+        package-manager:
+          required: true 
+          type: string
+        dependency-path:
+          required: true 
+          type: string
+        trigger-label:
+          required: false
+          type: string
+          default: none
+        path-doc:
+          required: false
+          type: string 
+          default: ./doc
+
+        
+          
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  check-label:
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.valid_label.outputs.value}}
+    steps:
+      - name: checks for the label 
+        id: valid_label
+        run: |
+          if [[ "${{ inputs.trigger-label }}" == "latest" ]] || [[ "${{ inputs.trigger-label  }}" == "stable" ]];
+            then
+                echo "value=true" >> $GITHUB_OUTPUT
+            else
+                echo "value=false" >> $GITHUB_OUTPUT 
```

Comment:
This one you do not need, just raise an error if it's false.
https://man7.org/linux/man-pages/man1/exit.1p.html
https://tldp.org/LDP/abs/html/exitcodes.html